src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseDefaultDllImportSearchPathsAttributeTests.cs

```diff
@@ -0,0 +1,380 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    // All the test cases use user32.dll as an example,
+    // however it is a commonly used system dll and will be influenced by Known Dlls mechanism,
+    // which will ignore all the configuration about the search algorithm.
+    // Fow now, this rule didn't take Known Dlls into consideration.
+    // If it is needed in the future, we can recover this rule.
+    public class UseDefaultDllImportSearchPathsAttributeTests : DiagnosticAnalyzerTestBase
+    {
+        // It will try to retrieve the MessageBox from user32.dll, which will be searched in a default order.
+        [Fact]
+        public void Test_DllImportAttribute_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Runtime.InteropServices;
+
+class TestClass
+{
+    [DllImport(""user32.dll"")]
+    public static extern int MessageBox(IntPtr hWnd, String text, String caption, uint type);
+
+    public void TestMethod()
+    {
+        MessageBox(new IntPtr(0), ""Hello World!"", ""Hello Dialog"", 0);
+    }
+}",
+            GetCSharpResultAt(8, 30, UseDefaultDllImportSearchPathsAttribute.Rule, "MessageBox"));
+        }
+
+        [Fact]
+        public void Test_DllInUpperCase_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Runtime.InteropServices;
+
+class TestClass
+{
+    [DllImport(""user32.DLL"")]
+    public static extern int MessageBox(IntPtr hWnd, String text, String caption, uint type);
+
+    public void TestMethod()
+    {
+        MessageBox(new IntPtr(0), ""Hello World!"", ""Hello Dialog"", 0);
+    }
+}",
+            GetCSharpResultAt(8, 30, UseDefaultDllImportSearchPathsAttribute.Rule, "MessageBox"));
+        }
+
+        [Fact]
+        public void Test_WithoutDllExtension_Diagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Runtime.InteropServices;
+
+class TestClass
+{
+    [DllImport(""user32"")]
+    public static extern int MessageBox(IntPtr hWnd, String text, String caption, uint type);
+
+    public void TestMethod()
+    {
+        MessageBox(new IntPtr(0), ""Hello World!"", ""Hello Dialog"", 0);
+    }
+}",
+            GetCSharpResultAt(8, 30, UseDefaultDllImportSearchPathsAttribute.Rule, "MessageBox"));
+        }
+
+        // It will have a compiler warning and recommend to use [DllImport]. So, there's no need to flag a diagnostic for this case.
```

Comment:
>It will have a compiler warning and recommend to use [DllImport]. So, there's no need to flag a diagnostic for this case. [](start = 11, length = 121)

No need to flag this case because no DLL will attempt to be loaded.